plugins/lua/orders.lua

```diff
@@ -709,11 +714,401 @@ function QuantityRightClickOverlay:onInput(keys)
     end
 end
 
+--
+-- OrdersSearchOverlay
+--
+
+local search_cursor_visible = false
+local search_last_scroll_position = -1
+
+local function make_order_key(order)
+    local mat_cat_str = ''
+    if order.material_category then
+        local keys = {}
+        for k in pairs(order.material_category) do
+            if type(k) == 'string' then
+                table.insert(keys, k)
+            end
+        end
+        table.sort(keys)
+        for _, k in ipairs(keys) do
+            mat_cat_str = mat_cat_str .. k .. '=' .. tostring(order.material_category[k]) .. ';'
+        end
+    end
+
+    local encrust_str = ''
+    if order.specflag and order.specflag.encrust_flags then
+        local flags = {'finished_goods', 'furniture', 'ammo'}
+        for _, flag in ipairs(flags) do
+            if order.specflag.encrust_flags[flag] then
+                encrust_str = encrust_str .. flag .. ';'
+            end
+        end
+    end
+
+    return string.format('%d:%d:%d:%d:%d:%s:%s:%s',
+        order.job_type,
+        order.item_type,
+        order.item_subtype,
+        order.mat_type,
+        order.mat_index,
+        order.reaction_name or '',
+        mat_cat_str,
+        encrust_str)
```

Comment:
this results in lexicographic ordering by the text representation of job type's number, which means 0, 1, 10, 11, 12, ..., 19, 2, 20, 21...? while this is a stable sort order, it is not an order the player will likely anticipate